### ISSUE TITLE
Not satisfaced dependencies on linux run

### ISSUE THREAD

**Issue by yhojann-cl** [cocos2d/cocos2d-x#17917]
- cocos2d-x version: 3
- devices test on: GNU/Linux (Ubuntu 16.04)
- developing environments
   - NDK version: -
   - Xcode version: -
   - VS version: -
   - browser type and version: -

Steps to Reproduce:

```
$ uname -a
Linux machine 4.4.0-79-generic #100-Ubuntu SMP Wed May 17 19:58:14 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
$ mkdir "~/Desktop/demo/" && cd "~/Desktop/demo/"
$ git clone https://github.com/cocos2d/cocos2d-x.git && cd cocos2d-x
$ python download-deps.py
$ git submodule update --init
$ ./setup.py
$ source /home/whk/.bashrc
$ cd build/
$ ./install-deps-linux.sh
$ sudo apt update
$ sudo apt upgrade
$ sudo apt install libxmu-dev libgl2ps-dev g++-4.9 libzip-dev libcurl4-gnutls-dev
$ cd ..
$ cocos new DemoProject -p cl.whk.demo -l js && cd DemoProject
$ cocos run -p linux
...
[100%] Linking CXX executable bin/MyGame
[100%] Built target MyGame
Build succeed.
Deploying mode: debug
Starting application.
running: '/home/whk/Desktop/demo/simulator/linux/MyGame'

/home/whk/Desktop/demo/simulator/linux/MyGame: error while loading shared libraries: libfmod.so.6: cannot open shared object file: No such file or directory
Error running command, return code: 127.
```
Y copied /home/whk/Desktop/demo/cocos2d-x/external/linux-specific/fmod/prebuilt/64-bit/* to /usr/share/libs/ and make symbolic link from libfmod.so to libfmod.so.6 but does not work, the app say same error.

Checking the dependencies:
```

Leyendo lista de paquetes... Hecho
Creando árbol de dependencias       
Leyendo la información de estado... Hecho
libglew-dev ya está en su versión más reciente (1.13.0-2).
libglu1-mesa-dev ya está en su versión más reciente (9.0.0-2.1).
libpng12-dev ya está en su versión más reciente (1.2.54-1ubuntu1).
fijado libpng12-dev como instalado manualmente.
libsqlite3-dev ya está en su versión más reciente (3.11.0-1ubuntu1).
libx11-dev ya está en su versión más reciente (2:1.6.3-1ubuntu2).
fijado libx11-dev como instalado manualmente.
libxi-dev ya está en su versión más reciente (2:1.7.6-1).
fijado libxi-dev como instalado manualmente.
libxmu-dev ya está en su versión más reciente (2:1.1.2-2).
libgl2ps-dev ya está en su versión más reciente (1.3.8-1.2).
libzip-dev ya está en su versión más reciente (1.0.1-0ubuntu1).
binutils ya está en su versión más reciente (2.26.1-1ubuntu1~16.04.3).
libcurl4-gnutls-dev ya está en su versión más reciente (7.47.0-1ubuntu2.2).
libfontconfig1-dev ya está en su versión más reciente (2.11.94-0ubuntu1.1).
fijado libfontconfig1-dev como instalado manualmente.
libssl-dev ya está en su versión más reciente (1.0.2g-1ubuntu4.8).
fijado libssl-dev como instalado manualmente.
libgtk-3-dev ya está en su versión más reciente (3.20.8-1ubuntu0~ppa1).
fijado libgtk-3-dev como instalado manualmente.
g++-4.9 ya está en su versión más reciente (4.9.4-2ubuntu1~16.04).
gcc-4.9 ya está en su versión más reciente (4.9.4-2ubuntu1~16.04).
fijado gcc-4.9 como instalado manualmente.
```
All dependencies are ok, but the application need more libraries.

**Comment by yhojann-cl** [comment]
The installer script need copy the libfmod libraries to user directory. The game need install libraries in client pc? i can copy libraries into binary folder? i can embed the libraries into binary game?

**Comment by Rubonnek** [comment]
That's true, the installer needs to, at least, create a symbolic link of the FMOD shared library.

I don't know if it's possible to embed the shared libraries due to licensing, perhaps one of the cocos2d-x developer might know, but you can always include a shell script with the release of the game and use something like:

```shell
LD_PRELOAD="/path/to/64-bit/libfmod.so" /path/to/my/game/binary
```

to load the FMOD library wherever they may be.

**Comment by Rubonnek** [comment]
The following symbolic links work in Arch Linux. They could work for Ubuntu but I have not tested that:

```shelll
sudo ln -sf "${COCOS_X_ROOT}"/external/linux-specific/fmod/prebuilt/64-bit/libfmod.so "/usr/lib/libfmod.so.6"
sudo ln -sf "${COCOS_X_ROOT}"/external/linux-specific/fmod/prebuilt/64-bit/libfmodL.so "/usr/lib/libfmodL.so.6"
```

**Comment by minggo** [comment]
I don't know why need to create a symbol link as `libfmod.so.6` as the codes just link to `libfmod`.

**Comment by Rubonnek** [comment]
According to `ldd` that's not true. This can be checked by compiling a brand new project and then:

```shell
ldd MyGame | fgrep libfmod
        libfmod.so.6 => /usr/lib/libfmod.so.6 (0x00007f1ae7b41000)
```

The shared library looked for is `libfmod.so.6`, not `libfmod.so`.

**Comment by Rubonnek** [comment]
I've also been maintaining the [`cocos2d-x-src`](https://aur.archlinux.org/packages/cocos2d-x-src) package in the Arch User Repository for more than a year and a half now and there hasn't been any issues related to FMOD reported by the users during this time.

The package patches cocos2d-x, but does not touch the build process at all.

**Comment by minggo** [comment]
@Rubonnek yep, i don't know why it search for libfmod.so.6 instead of libfmod.so. I tried to find the reason and failed long ago.

And it is interesting that you maintained cocos2d-x in Arch User Repository. What does it been used for?

**Comment by Rubonnek** [comment]
@minggo Apart from all my prototypes, I honestly don't know. I don't know what the Arch Linux users are using cocos2d-x for, and I don't know either how many developers are using cocos2d-x in Arch Linux.

I've mostly used used it to polish my C and C++ knowledge by poking around the engine and finding and porting a bunch of old code found online like my [AnimatedLabel](https://github.com/Rubonnek/animatedlabel/blob/master/Classes/AnimatedLabel.h) class that I forked ported over to the latest version of cocos2d-x from [CCLabelBMFontAnimated](https://github.com/SteveBarnegren/CCLabelBMFontAnimated) ([demo video](https://www.youtube.com/watch?v=Ku8kq8CY_vk) by @SteveBarnegren), and also porting an [old dear imgui integration to cocos2d-x version v3.14](https://github.com/Rubonnek/imguix). I also added encryption support for some extra files with XXTEA and included the wxsqlite database into my personal projects.
